domain/consensus/processes/blockprocessor/validate_and_insert_block.go

```diff
@@ -134,6 +134,21 @@ func (bp *blockProcessor) validateAndInsertBlock(stagingArea *model.StagingArea,
 		}
 	}
 
+	err = bp.blockValidator.ValidateHeaderPruningPoint(stagingArea, blockHash)
```

Comment:
Because it's done after we move the pruning point. I can move it inside validateBlock, but then we would need to call `NextPruningPointAndCandidateByBlockHash` because the new pruning point is not set yet. What do you think?